### ISSUE TITLE
Remove Package Naming Requirement

### ISSUE THREAD

**Issue by Nytelife26** [eslint/eslint#14034]
**The version of ESLint you are using.**
7.18.0

**The problem you want to solve.**
As of present, scoped packages MUST be prefixed with `eslint-config-`. This is understandable for general NPM packages (somewhat, still should not be a requirement if people so desire to name it differently), but as for scoped packages where organizations can ensure there will be no naming conflicts, this is unreasonable.

**Your take on the correct solution to the problem.**
It should be a *recommendation*, not a requirement. ESLint should still allow you to use any other package name, as it is not the position of the utility to impose organizational structure.

**Are you willing to submit a pull request to implement this change?**
Yes.

**Comment by ljharb** [comment]
Given that there's no issue with finding a name inside a scope, what's wrong with naming it `@scope/eslint-config`?

**Comment by mdjermanovic** [comment]
Hi @Nytelife26, thanks for the issue! 

> As of present, scoped packages MUST be prefixed with `eslint-config-`.

That is a convention if the config is specified by its name in `extends`, but absolute paths are also allowed so you can specify the shared config in `.eslintrc.js` like this:

```js
module.exports = {
    extends: [require.resolve("@myscope/any-package-name")]
};
```

> It should be a _recommendation_, not a requirement. ESLint should still allow you to use any other package name, as it is not the position of the utility to impose organizational structure.

We are already working on the new config format per eslint/rfcs#9, where shared configs should be `require`d directly from `eslint.config.js`, so there will be no restrictions or assumptions about the config's name. 

You can follow the progress here: #13481.

**Comment by nzakas** [comment]
As already noted, this restrictions is already being removed with eslint/rfcs#9, so closing.

**Comment by Nytelife26** [comment]
> Hi @Nytelife26, thanks for the issue!
> 
> > As of present, scoped packages MUST be prefixed with `eslint-config-`.
> 
> That is a convention if the config is specified by its name in `extends`, but absolute paths are also allowed so you can specify the shared config in `.eslintrc.js` like this:

This is actually not the case. ESLint does not register nor recognize packages, even scoped ones, unless they are prefixed with `eslint-config-`.

**Comment by mdjermanovic** [comment]
> > That is a convention if the config is specified by its name in `extends`, but absolute paths are also allowed so you can specify the shared config in `.eslintrc.js` like this:
> 
> This is actually not the case. ESLint does not register nor recognize packages, even scoped ones, unless they are prefixed with `eslint-config-`.

Did you try with [require.resolve](https://github.com/eslint/eslint/issues/14034#issuecomment-766542136)? It should work with any package name.
